src/test/scala/org/constellation/snapshotstreaming/CurrencySnapshotMapperSuite.scala

```diff
@@ -0,0 +1,274 @@
+
+    val rewards = createRewards(address1, address2, address7)
+    for {
+      txn1 <- createTxn(address1, key1, address2)
+      txn2 <- createTxn(address2, key2, address1)
+      txn3 <- createTxn(address2, key2, address7)
+
+      feeTxn1 <- createFeeTxn(address3, key3, address4)
+      feeTxn2 <- createFeeTxn(address4, key4, address3)
+      feeTxn3 <- createFeeTxn(address4, key4, address7)
+
+      blocks <- createBlocksWithTransactions(
+        key1,
+        NonEmptySet.fromSetUnsafe(SortedSet(txn1, txn2)),
+        NonEmptySet.fromSetUnsafe(SortedSet(txn3))
+      )
+      feeTransactions = SortedSet(feeTxn1, feeTxn2, feeTxn3).some
+      snapshot <- incrementalCurrencySnapshot[IO](
+        100L,
+        10L,
+        20L,
+        Hash("abc"),
+        Hash("def"),
+        totalInfo,
+        blocks,
+        rewards = rewards,
+        feeTransactions = feeTransactions
+      )
+
+      result = CurrencyIncrementalSnapshotMapper.make().snapshotReferredBalancesInfo(snapshot, totalInfo)
+    } yield expect.same(
+      result,
+      SortedMap(
+        address1 -> Balance(0L),
+        address2 -> Balance(0L),
+        address3 -> Balance(0L),
+        address4 -> Balance(0L))
+    )
+
+  }
+
+  test("leave balances for addresses from transactions and fee transactions") { res =>
+    implicit val (h, ks, sp, key1, key2, key3, key4) = res
+    val address1 = key1.getPublic.toAddress
+    val address2 = key2.getPublic.toAddress
+    val address3 = key3.getPublic.toAddress
+    val address4 = key4.getPublic.toAddress
+    val balances = createBalances(address1, address2, address3, address4, address5, address6, address7)
+    val totalInfo = CurrencySnapshotInfo(SortedMap.empty, balances, None, None)
+
+    for {
+      txn1 <- createTxn(address1, key1, address2)
+      txn2 <- createTxn(address2, key2, address1)
+      txn3 <- createTxn(address2, key2, address5)
+      feeTxn1 <- createFeeTxn(address3, key3, address4)
+      feeTxn2 <- createFeeTxn(address4, key4, address3)
+      feeTxn3 <- createFeeTxn(address4, key4, address7)
+      feeTransactions = SortedSet(feeTxn1, feeTxn2, feeTxn3).some
+      blocks <- createBlocksWithTransactions(
+        key1,
+        NonEmptySet.fromSetUnsafe(SortedSet(txn1, txn2)),
+        NonEmptySet.fromSetUnsafe(SortedSet(txn3))
+      )
+      snapshot <- incrementalCurrencySnapshot[IO](
+        100L,
+        10L,
+        20L,
+        Hash("abc"),
+        Hash("def"),
+        totalInfo,
+        blocks,
+        feeTransactions = feeTransactions
+      )
+
+      result = CurrencyIncrementalSnapshotMapper.make().snapshotReferredBalancesInfo(snapshot, totalInfo)
+      expectedBalances = createBalances(address1, address2, address3, address4, address5, address7)
+    } yield expect.same(result, expectedBalances)
+  }
+
+  test(
+    "leave balances for addresses from transactions and fee transactions, but not set zero for destinations not in balances "
```

Comment:
How is it that two tests with the same name pass? 🤔

Reply:
Because we have some common test cases for both Global and Currency mappers